### PR TITLE
Adopt package-lock.json

### PR DESCRIPTION
**Pull request by amcasey**
Add a workflow to update it daily overnight.

### CODE REVIEW

**Comment by amcasey** [comment]
I believe the AzDO pipelines already respect this since they share a task group with the release branches.

File: .github/workflows/update-package-lock.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v1
+      with:
+        node-version: 12
+        registry-url: https://registry.npmjs.org/
+
+    - name: Configure git and update package-lock.json
+      run: |
+        git config user.email "ts_bot@rcavanaugh.com"
```
Comment (amcasey) [inline .github/workflows/update-package-lock.yaml:22]: @RyanCavanaugh This is the right email address for the bot?

File: .github/workflows/update-package-lock.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Update package-lock.json
```
Comment (amcasey) [inline .github/workflows/update-package-lock.yaml:1]: @weswigham @andrewbranch Is there a better way to test this than confirming it ran the night after this PR merged?

File: .gitignore [hunk 3]
```diff
@@ -84,7 +84,6 @@ tests/cases/user/*/**/*.d.ts
 tests/baselines/reference/dt
 .failed-tests
 TEST-results.xml
-package-lock.json
```
Comment (weswigham) [inline .gitignore:87]: This should stay, so local changes (if there are any) aren't added by a `git add .`. The auto-update script should forcibly add the lockfile despite the ignore.

File: .github/workflows/update-package-lock.yaml (hunk 2 above)
Comment (andrewbranch) [reply .github/workflows/update-package-lock.yaml:1]: Probably, but I don’t think any of them are as good as the way you mentioned. You can add a `workflow_dispatch` trigger in addition to the `schedule` trigger to be able to run it manually.

File: .github/workflows/update-package-lock.yaml (hunk 1 above)
Comment (andrewbranch) [reply .github/workflows/update-package-lock.yaml:22]: typescriptbot@microsoft.com (see https://github.com/typescript-bot)

File: .gitignore (hunk 3 above)
Comment (amcasey) [reply .gitignore]: Great idea!

File: .github/workflows/update-package-lock.yaml (hunk 1 above)
Comment (amcasey) [reply .github/workflows/update-package-lock.yaml:22]: I'll go ahead and update the other workflows?

Comment (amcasey) [reply .github/workflows/update-package-lock.yaml]: Done.  Shout if that's a problem.

Comment (weswigham) [reply .github/workflows/update-package-lock.yaml]: Its avatar is associated with that address on https://en.gravatar.com/ right? That's really the only visible part.

Comment (amcasey) [reply .github/workflows/update-package-lock.yaml]: @weswigham which address is on gravatar?

Comment (weswigham) [reply .github/workflows/update-package-lock.yaml]: The email associated with the commit. GitHub uses gravatar to find an avatar for it to display on the GitHub UI.

**Pull request merged**